### PR TITLE
Cloning Rework

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Yeah I revived the old fishfish pr, draft because I still have a lot of cleanup and I can't merge upstream because it crashes with a weightless bug.

**Changelog**

:cl: Fishfish458, Rane
- tweak: Reworked cloning. It now takes a cloning pod, cloning console, and medical scanner linked together with a multitool.
- remove: You can no longer prescan to be cloned. Your body has to be in the scanner at the time of cloning.

### CODE REVIEW

**Comment by Elijahrane** [comment]
What do you want me to do about body reassemble since it's already pretty broken and incompatible with this

**Comment by EmoGarbage404** [comment]
> What do you want me to do about body reassemble since it's already pretty broken and incompatible with this

Um you can probably leave it commented out and then i can fix it after the pr is merged. On thing of note, im not sure the species field in humanoid appearance is properly updated and tracked.

**Comment by EmoGarbage404** [comment]
Ive made fixes to skeletons in a new pr, so i suggest removing all unnecessary changes in your code lest you may run into a lot of conflicts.

**Comment by Elijahrane** [comment]
I'll just fix it after that's merged because this isn't gonna build if I uncomment that

File: Content.Server/Body/Components/BodyReassembleComponent.cs
Comment (mirrorcult) [inline Content.Server/Body/Components/BodyReassembleComponent.cs:34]: has this been discussed with emogarbage at all as for whether this is getting fixed soon, cause ideally this is uncommented before merge

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:73]:
```suggestion
```

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:76]:
```suggestion
```

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:83]:
```suggestion
```

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:175]:
```suggestion
                Transform(scanner.Value).Coordinates.TryDistance(EntityManager, Transform((console)).Coordinates, out float scannerDistance);
                consoleComp.GeneticScannerInRange = scannerDistance <= consoleComp.MaxDistance;
            }
            if (cloningPod != null)
            {
                Transform(cloningPod.Value).Coordinates.TryDistance(EntityManager, Transform((console)).Coordinates, out float podDistance);
                consoleComp.CloningPodInRange = podDistance <= consoleComp.MaxDistance;
```

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:185]: localize

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:95]: these should be static strings/consts somewhere

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:217]: loc

File: Content.Server/Cloning/CloningConsoleSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningConsoleSystem.cs:225]:
```suggestion
            && Transform(consoleComponent.CloningPod.Value).Anchored)
```
etc for all other cases of that

File: Content.Server/Cloning/CloningSystem.cs
Comment (mirrorcult) [inline Content.Server/Cloning/CloningSystem.cs:164]: this should really use an `ActiveCloningPodComponent` for perf

File: Content.Server/Cloning/Components/CloningConsoleComponent.cs
Comment (mirrorcult) [inline Content.Server/Cloning/Components/CloningConsoleComponent.cs:35]: this doesnt need to be stateful

File: Content.Shared/Cloning/SharedCloningConsole.cs
Comment (mirrorcult) [inline Content.Shared/Cloning/SharedCloningConsole.cs:9]: entire file is indented one too many

**Review by mirrorcult — CHANGES_REQUESTED** [review]
will need another review + i need to test ingame as well

**Comment by Elijahrane** [comment]
image

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**
